### PR TITLE
Update CARTO API description #17

### PR DESCRIPTION
**Pull request by jaakla**
Added 2 so far missing, but important API integration docs to Mobile API
@csobier please review, and can you push it to a staging?

### CODE REVIEW

File: doc/sdk_with_carto.md [hunk 1]
```diff
@@ -55,10 +55,602 @@ This high-level workflow describes how to prepare your mobile data for rendering
...
+
+Reference: https://carto.com/docs/carto-engine/carto-js/
 
 In order to integrate your published mobile map into your app, you need a callback-based asynchronous process to request the viz.json.
 
+Warning! viz.json API is in development and experimental. Use at your own risk.
+
 1)  The following requirements are needed for app integration of a published CARTO Mobile SDK map:
 
   - Online API is requested and will run in another thread
```
Comment (csobier) [inline doc/sdk_with_carto.md]: What is the "online API"? 
And how is this section different from https://carto.com/docs/carto-engine/mobile-sdk/sdk-with-other-carto-tools/#sdk-and-carto-apis (that starts on line 5-9 of this file)?

**Comment by csobier** [comment]
@jaakla and @Nikituh , looks good. I just applied some minor text updates. Can you answer the questions above? Then I can push and create a Docs staging to publish the live version.

File: doc/sdk_with_carto.md [hunk 2]
```diff
@@ -55,10 +55,602 @@ This high-level workflow describes how to prepare your mobile data for rendering
...
+</div>
+
+### CARTO.js API
+
+Reference: https://carto.com/docs/carto-engine/carto-js/
 
 In order to integrate your published mobile map into your app, you need a callback-based asynchronous process to request the viz.json.
 
+Warning! viz.json API is in development and experimental. Use at your own risk.
+
```
Comment (jaakla) [inline doc/sdk_with_carto.md]: yes

File: doc/sdk_with_carto.md (hunk 1 above)
Comment (jaakla) [inline doc/sdk_with_carto.md]: Online API is general term here, maybe better would be "Network request is done and this has to run in own thread.". 

This section was here before also, just moved to end of the document, as the carto.js API is most unstable and is not suggested as the first integration option.

File: doc/sdk_with_carto.md [hunk 3]
```diff
@@ -2,6 +2,40 @@
...
+3) Click _SHARE_ to publish your map
+
+4) Select _CARTO Mobile SDK_ as the publish option
+
+  This publishing option provides a URL to the mobile viz.json file, which is required if you are using the CARTO Mobile SDK to publish custom maps for Android, iOS, and Windows platforms.
+
+  **Note:** [Widgets](https://carto.com/learn/guides/widgets/exploring-widgets) are currently not supported for the CARTO Mobile SDK format, but will be available in a future release.
+
 ## SDK and CARTO APIs
```
Comment (csobier) [inline doc/sdk_with_carto.md]: @jaakla , I moved this section down. Is that what you wanted to do? 
Please ensure that this is not dup content of line #639.

**Comment by jaakla** [comment]
Looks good to me. @csobier will you merge it, right?

**Comment by csobier** [comment]
Great! Yes, will merge now and just need to run a script in a Docs PR to publish the changes. Doing it now. thanks!

**Pull request merged**